integration-tests/chopsticks/src/tests/xcm/types.ts

```diff
@@ -0,0 +1,48 @@
+import type { ApiPromise } from '@polkadot/api'
+import type { SubmittableExtrinsic } from '@polkadot/api/types'
+import type { KeyringPair } from '@polkadot/keyring/types'
+import type { SetupOption, EventFilter } from '@acala-network/chopsticks-testing'
+
+export interface Storage {
+	senderStorage: Record<string, Record<string, unknown>>
+	receiverStorage: Record<string, Record<string, unknown>>
+	relayStorage: Record<string, Record<string, unknown>>
+}
+
+export interface Accounts {
+	senderAccount: KeyringPair
+	receiverAccount: KeyringPair
+}
+
+export interface NetworkSetupOption {
+	sender: SetupOption
+	receiver: SetupOption
+	relay: SetupOption
+}
+
+export interface BasisTxContext {
```

Comment:
You mean

```suggestion
export interface BasicTxContext {
```

?

Reply:
Also, same reasoning about a more generic approach at least for the `pallets` property applies.